### PR TITLE
DEX-803 make view buttons go to collaboration table directly

### PR DESCRIPTION
**Pull request by Atticus29**
Addresses JH's QA feedback for DEX-803:

> Following up on our Slack conversation. I think all that is left is the make the notifications for a collaboration link to the collaboration table when the “View” button is clicked.

I tried to follow Emily's pattern from her recent PR and use useLocation. I don't think that it fit exactly my use case.
Instead, I followed some online advice(https://dev.to/gedalyakrycer/5-remarkable-react-router-features-anchor-links-query-params-more-2aeg#anchorLinks and https://stackoverflow.com/questions/40280369/use-anchors-with-react-router) and added the 'react-router-hash-link' package.

Slightly added to the ButtonLink component to make the use of HashLinks from within buttons easier going forward.

### CODE REVIEW

File: src/components/UserProfile.jsx [hunk 1]
```diff
@@ -174,7 +174,12 @@ export default function UserProfile({
               someoneElse ? 'NO_SIGHTINGS_NON_SELF' : 'NO_SIGHTINGS'
             }
           />
-          {!someoneElse && <CollaborationsCard userId={userId} />}
+          {!someoneElse && (
+            <CollaborationsCard
+              htmlId="collab-card"
```
Comment (Atticus29) [inline src/components/UserProfile.jsx:179]: Oh yeah. I forgot that I also propagated the necessary element id attribute down to the material element.

File: src/components/ButtonLink.jsx [hunk 2]
```diff
@@ -7,9 +8,21 @@ export default function ButtonLink({
...
+  isHashLink = false,
   linkProps,
   ...rest
 }) {
+  if (isHashLink) {
+    return (
+      <HashLink
+        style={{ textDecoration: 'unset' }}
+        to={href}
+        {...linkProps}
```
Comment (Emily-Ke) [inline src/components/ButtonLink.jsx:20]: It doesn't seem to hurt anything, but HashLink is probably not expecting these props.

**Review by Emily-Ke — APPROVED** [review]
One minor comment.

**Review by brmscheiner — APPROVED** [review]
very nice!

**Pull request merged**